sacred/observers/file_storage.py

```diff
@@ -51,16 +50,42 @@ def __init__(self, basedir, resource_dir, source_dir, template,
         self.cout = ""
         self.cout_write_cursor = 0
 
-    def queued_event(self, ex_info, command, host_info, queue_time, config,
-                     meta_info, _id):
-        if not os.path.exists(self.basedir):
-            os.makedirs(self.basedir)
+    @staticmethod
+    def _makedirs(name, mode=0o777, exist_ok=False):
+        """ Wrapper of os.makedirs with fallback
+            for exist_ok on python 2.
+        """
+        try:
+            os.makedirs(name, mode, exist_ok)
+        except TypeError:
+            if not os.path.exists(name):
+                os.makedirs(name, mode)
+
+    def _make_run_dir(self, _id):
+        self._makedirs(self.basedir, exist_ok=True)
         if _id is None:
-            self.dir = tempfile.mkdtemp(prefix='run_', dir=self.basedir)
```

Comment:
This whole block could be placed in its own method to increase readability.

Reply:
I separated to a new method. Please review whether it still works like you expect.